meeshkan/serve/mock/specs.py

```diff
@@ -16,27 +18,51 @@ class OpenAPISpecification:
     source: str
 
 
-def load_specs(specs_dir: str) -> Sequence[OpenAPISpecification]:
-    if not os.path.exists(specs_dir):
-        logging.info("OpenAPI schema directory not found %s", specs_dir)
-        return []
+def load_specs(specs: Union[str, Sequence[str]]) -> Sequence[OpenAPISpecification]:
+    if isinstance(specs, str):
+        specs = (specs,)
+
     specs_with_sources = []
-    specs_paths = [
-        s
-        for s in os.listdir(specs_dir)
-        if s.endswith("yml") or s.endswith("yaml") or s.endswith("json")
-    ]
-    for spec_path in specs_paths:
-        full_spec_path = os.path.join(specs_dir, spec_path)
-        with open(full_spec_path, encoding="utf8") as spec_file:
+
+    for spec in specs:
+        is_http = spec.startswith("http://") or spec.startswith("https://")
+
+        if is_http or os.path.isfile(spec):
+            specs_paths = [spec]
+        elif os.path.isdir(spec):
+            specs_paths = [
+                os.path.join(spec, dir_entry)
+                for dir_entry in os.listdir(spec)
+                if dir_entry.endswith("yml")
+                or dir_entry.endswith("yaml")
+                or dir_entry.endswith("json")
+            ]
+        else:
+            raise ClickException(f"OpenAPI specification {spec} not found")
+
+        for spec_source in specs_paths:
+            spec_text: str
+            if is_http:
+                try:
+                    response = requests.get(spec)
```

Comment:
How's this mocked in tests, I guess by patching? Would it be better to create some kind of a `SpecLoader` class for this 🤔So that one could also externalize the `requests.get` away from here?

Reply:
Haven't added a test yet - will do!

Reply:
A test using `requests_mock` (we already used that in another place) has now been added! I don't think we need to introduce an abstraction at this stage (it might come later as we're introducing additional support).